packages/pds/src/pipethrough.ts

```diff
@@ -1,125 +1,75 @@
 import express from 'express'
 import * as ui8 from 'uint8arrays'
 import net from 'node:net'
-import { LexValue, jsonToLex, stringifyLex } from '@atproto/lexicon'
-import { HandlerPipeThrough, InvalidRequestError } from '@atproto/xrpc-server'
+import stream from 'node:stream'
+import webStream from 'node:stream/web'
+import { jsonToLex } from '@atproto/lexicon'
+import {
+  CatchallHandler,
+  HandlerPipeThrough,
+  InvalidRequestError,
+} from '@atproto/xrpc-server'
 import { ResponseType, XRPCError } from '@atproto/xrpc'
 import { ids, lexicons } from './lexicon/lexicons'
 import { httpLogger } from './logger'
 import { getServiceEndpoint, noUndefinedVals } from '@atproto/common'
 import AppContext from './context'
 
-type PipethroughOptions = {
-  /**
-   * Request headers to pass-through, in addition to those defined in
-   * {@link REQ_HEADERS_TO_FORWARD}
-   */
-  reqHeadersToForward?: string[]
-}
-
-const defaultService = (
-  ctx: AppContext,
-  path: string,
-): { url: string; did: string } | null => {
-  const nsid = path.replace('/xrpc/', '')
-  switch (nsid) {
-    case ids.ToolsOzoneCommunicationCreateTemplate:
-    case ids.ToolsOzoneCommunicationDeleteTemplate:
-    case ids.ToolsOzoneCommunicationUpdateTemplate:
-    case ids.ToolsOzoneCommunicationListTemplates:
-    case ids.ToolsOzoneModerationEmitEvent:
-    case ids.ToolsOzoneModerationGetEvent:
-    case ids.ToolsOzoneModerationGetRecord:
-    case ids.ToolsOzoneModerationGetRepo:
-    case ids.ToolsOzoneModerationQueryEvents:
-    case ids.ToolsOzoneModerationQueryStatuses:
-    case ids.ToolsOzoneModerationSearchRepos:
-      return ctx.cfg.modService
-    case ids.ComAtprotoModerationCreateReport:
-      return ctx.cfg.reportService
-    default:
-      return ctx.cfg.bskyAppView
+export const proxyHandler =
+  (ctx: AppContext): CatchallHandler =>
+  async (req, res, next) => {
+    try {
+      const auth = await ctx.authVerifier.access({ req })
```

Comment:
It would be kind of nice to bail early and not bother with the work of auth'ing if the request is non-lexrpc.

Reply:
nice - i split up the `createUrlAndHeaders` function and now do url + audience before checking auth and then create the signed headers after checking auth 👌